src/runtime/ranger/ranger_resource_policy.h

```diff
@@ -46,12 +47,18 @@ extern access_type operator|(access_type lhs, access_type rhs);
 
 extern access_type operator&(access_type lhs, access_type rhs);
 
+extern access_type &operator|=(access_type &lhs, access_type rhs);
+
+extern uint8_t access_type_to_int8_t(const access_type &ac_type);
+
 // Ranger policy data structure
 struct policy_item
 {
-    access_type access_types;
+    uint8_t access_types;
```

Comment:
I guess you changed it from `access_type` to `uint8_t` is because that enum class is not convenient in same cases, right?

Reply:
Yes, when serializing like access_type::KRead | access_type::KWrite, will fail

Reply:
I found you have overrided some operators, e.g. `|`, `&`, and etc, so where else needed to use `uint8_t`?

@empiredan do you think it's worth to use enum class? It seems there are too many static_casts.

Reply:
For example: I need to store `access_type::KRead | access_type::KWrite` in zk, but such a combination cannot be serialized correctly, it will be serialized as `unknown`, if I deserialize it, it will lose `access_type::KRead | access_type::KWrite` meaning, so I need to convert it to `uint8_t`.
mainly after c++11, it is recommended to use enum class. If it can handle the logical relationship well, I think it is no problem. What do you think, hope your reply